### DISCUSSION TITLE
📱 Mobile App Version (computer / mobile sync)

### DISCUSSION THREAD

**Discussion by gogl92** [diegopvlk/Dosage#59] [Announcements]
I'm not sure if this makes sense but it would be awesome if I can register my medications in the Dosage and export them to a calendar event so even if I'm not in my computer I can receive the alert.

The app can easily create a calendar link event like this site does https://www.labnol.org/calendar/ but will create the issue of duplicated notifications, other option may be to be able to "sync" with those services to cancel or edit the event (or task https://support.google.com/tasks/answer/7675772?hl=en).

Please let me know your thoughts.

Thank you

**Comment by diegopvlk** [comment]
It does make sense and it's something that I want as well but it's not the right approach.

A mobile version of the app that syncs with the computer would be better. I already started making it, just don't have a time-frame for when it's gonna be ready.

**Comment by gogl92** [comment]
@diegopvlk that's sounds good! let me know if I can help

**Comment by diegopvlk** [comment]
Thank you, but it's something that I need to do by myself so I can learn :)

**Comment by reesericci** [comment]
I'm looking for a way to sync this with my apple healthkit data, which maybe a mobile app could be useful for this.

**Comment by diegopvlk** [comment]
> I'm looking for a way to sync this with my apple healthkit data, which maybe a mobile app could be useful for this.

No idea how this works, I also don't have an iPhone/Mac. So, very unlikely.

**Comment by diegopvlk** [comment]
There still a lot to do but I want to show a preview:

![Screenshot](https://github.com/diegopvlk/Dosage/assets/25941156/be1d53aa-0af6-48e3-8db4-c46a79b8cd97)

**Comment by HammyHavoc** [comment]
Just wanted to say this looks absolutely phenomenal and is looking like a great alternative to Medisafe.

**Comment by paulle** [comment]
Once the mobile app is ready will you offer in f-droid (too) ?

**Comment by diegopvlk** [comment]
> Once the mobile app is ready will you offer in f-droid (too) ?

It'll will be available first on play store, only because the [requirement](https://medium.com/welltested-ai/new-google-playstores-20-testers-policy-for-production-release-where-are-they-going-with-it-7e9c924464a9) of 20 testers. After that it'll be on f-droid.

**Comment by paulle** [comment]
Gesendet von Proton Mail für Mobilgeräte

Thank you

-------- Original-Nachricht --------
Am 5. Feb. 2024, 12:42, schrieb Diego Povliuk :

**Comment by Admiralfeb** [comment]
Looking forward to it!
I've been considering adding similar functionality into Home Assistant to improve my medication management considering I'm having a bit of issue with medisafe.

**Comment by HammyHavoc** [comment]
> Looking forward to it!
> I've been considering adding similar functionality into Home Assistant to improve my medication management considering I'm having a bit of issue with medisafe.

Dosage integration on HomeAssistant would be absolutely brilliant and solve so many problems for me and the in-laws.

**Comment by LinuxAndCoffee** [comment]
> > Once the mobile app is ready will you offer in f-droid (too) ?
> 
> It'll will be available first on play store, only because the [requirement](https://medium.com/welltested-ai/new-google-playstores-20-testers-policy-for-production-release-where-are-they-going-with-it-7e9c924464a9) of 20 testers. After that it'll be on f-droid.

I'd be happy to be a tester for you for the mobile app (and for anything testing with the Linux app). Thank you for making this, I cannot wait to start using it!!

**Comment by paulle** [comment]
I would like to be a tester too.

Gesendet von Proton Mail für Mobilgeräte

-------- Original-Nachricht --------
Am 10. März 2024, 02:04, schrieb LinuxAndCoffee :

**Comment by gaussandhisgun** [comment]
Where does one apply to test the mobile version? This looks like it could use a person who knows how to capture crash logs :D

**Comment by diegopvlk** [comment]
It's not ready yet. When it is I'll announce here.

**Comment by ada-magicat** [comment]
How do you plan on implementing the sync between devices?

**Reply by gogl92** [reply]
I was thinking about it, it may need some login/authentication mechanism too keep all devices synced.

**Comment by akhilerm** [comment]
Is there an update on the mobile version?

**Comment by diegopvlk** [comment]
Hi everyone and sorry to disappoint you all, I was rewriting the app in flutter for multiplatform support, but I didn't finish for lack of time and a bit of interest.

> How do you plan on implementing the sync between devices?

That's the hardest part. I was planning on using Firebase (not happy about it), I even published it [here](https://dosage-276e2.firebaseapp.com/), and yes, it would require login.
What I really wanted was something without a central server, if anyone has some recommendations I'm happy to hear, preferably something that will not cost me any money since I don't make any from this app.

I'll probably have to do some big changes in the code as well since that was a while ago and flutter changes rapidly and no idea when I'll have time for this, so I apologize for getting your hopes up :(

**Reply by HammyHavoc** [reply]
> Hi everyone and sorry to disappoint you all, I was rewriting the app in flutter for multiplatform support, but I didn't finish for lack of time and a bit of interest.
> 
> > How do you plan on implementing the sync between devices?
> 
> That's the hardest part. I was planning on using Firebase (not happy about it), I even published it [here](https://dosage-276e2.firebaseapp.com/), and yes, it would require login.
> What I really wanted was something without a central server, if anyone has some recommendations I'm happy to hear, preferably something that will not cost me any money since I don't make any from this app.
> 
> I'll probably have to do some big changes in the code as well since that was a while ago and flutter changes rapidly and no idea when I'll have time for this, so I apologize for getting your hopes up :(

In terms of solutions, Postgresql is always great. Failing that, WebDAV support is nice and it can hook up to existing sync solutions people have, e.g., Nextcloud. SQLite with a Docker container for access to it could also work. Users could even host their own SQLite database on one of Cloudflare's free offerings.

Matrix protocol could also be cool as it's about far more than just comms. Nordeck's matrix-neoboard project is very impressive—real-time collaborative whiteboard via Matrix protocol that also offers storage of the data via Matrix too. Might be interesting to reference their repo for inspiration.

If you're unsure about auth systems, consider offering endpoints for the likes of Authentik. But with that said, WebDAV would solve your problem with both auth, storage and sync. Plenty of reference examples available too for WebDAV.

Browser extensions, like ViolentMonkey, offer sync via WebDAV. Works well.

Here's a further weird idea: a Git repository for sync, but it's probably inappropriate. It would at least keep version history though so if a user has a problem then they can mirror their repo elsewhere or roll back to the last good commit.

Just some ideas!

**Reply by ada-magicat** [reply]
I want to recommend p2panda for the sync. It is a local first solution (no login, no central server) There's already collaboration in the gnome community with them:

https://blogs.gnome.org/tbernard/2023/11/23/local-first-workshop-feat-p2panda/
https://blogs.gnome.org/tbernard/2024/12/18/aardvark/
https://blogs.gnome.org/tbernard/2025/06/30/aardvark-summer-2025-update/

**Reply by diegopvlk** [reply]
@HammyHavoc Very cool ideas but a bit complicated to setup for end users I'm afraid, needs to be simpler.

> Here's a further weird idea: a Git repository for sync

I thought about using secret github gists, don't know if it's a good idea :P

> I want to recommend p2panda for the sync. It is a local first solution (no login, no central server)

@ada-magicat That's seems to be what I want, I'll read more about it.
